src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -606,20 +625,14 @@ private void onGameInfo(GameInfoMessage gameInfoMessage) {
 
     Game game = createOrUpdateGame(gameInfoMessage);
     if (GameStatus.CLOSED == game.getStatus()) {
-      if (!currentPlayerOptional.isPresent() || currentPlayerOptional.get().getGame() != game) {
-        removeGame(gameInfoMessage);
+      removeGame(gameInfoMessage);
+      if (!currentPlayerOptional.isPresent() || !Objects.equals(currentGame.get(), game)) {
         return;
       }
+      synchronized (currentGame) {
+        currentGame.set(null);
+      }
 
-      // Don't remove the game until the current player closed it. TODO: Why?
```

Comment:
I made a lot of research in the code and this listener causes a lot of issues and yet the comment that the game should only removed when the game process is closed makes absolutely no sense. The game process closing does not trigger this variable to change. Also for it to change we need to call removeGame(gameInfoMessage); .... So it will get stuck there never removing the current game... The code handeling game from the server is a little buggy as I noticed. I hope the restrucuring rules out most of them